utils: Improve debug message

BNieuwenhuizen opened the pull request:
This print the full directory path up to and including cur, because after each iteration we restore the slash.

I had some problems with /dev/vboxusb permissions and an error message like

crun: mkdir `009`: Permission denied: OCI permission denied

isn't too useful while

crun: mkdir `dev/vboxusb/009`: Permission denied: OCI permission denied

actually pointed me where to look.

rhatdan commented:
LGTM

giuseppe commented:
I am 50/50 on this change since the previous version was showing the subpath that failed instead of the entire directory name, but since you find it more useful I am fine to go with this version.  Just a nit, could you make sure the printed directory is absolute?  You may need to use `/%s` instead of `%s` so the `/`

BNieuwenhuizen commented:
> I am 50/50 on this change since the previous version was showing the subpath that failed instead of the entire directory name, but since you find it more useful I am fine to go with this version. Just a nit, could you make sure the printed directory is absolute? You may need to use `/%s` instead of `%s` so the `/`

Done. Note that npath puts a 0 at the end of the current path component, so this still gives you the exact part of the path that failed and hence strictly more information. (though I can see how one would miss that if not aware)

flouthoc reviewed (approved):
LGTM @giuseppe PTAL

giuseppe reviewed (approved):
LGTM
